book_config.py

```diff
@@ -195,8 +189,13 @@ def search_for_goodreads_url(self):
             self.goodreads_url_edit.setText('')
 
     def update_aliases(self):
+        if 'goodreads.com' not in self.goodreads_url_edit.text():
```

Comment:
I think urlsplit works by searching for www and if it's not there, then it
doesn't put into netloc.. So we'd essentially end up searching for '
goodreads.com' in ''.. You can see it by doing urlpase(urlsplit('
goodreads.com/test')) -- reason I didn't wanna do it this way is because
that url is still valid but would fail for us

On Wed, Jun 29, 2016, 5:12 PM Samreen Zarroug <szarroug3@gmail.com> wrote:

> Yeah
>
> On Wed, Jun 29, 2016, 3:51 PM Anthony Toole <notifications@github.com>
> wrote:
>
>> In book_config.py
>> <https://github.com/szarroug3/X-Ray_Calibre_Plugin/pull/55#discussion_r69025248>
>> :
>>
>> > @@ -195,8 +189,13 @@ def search_for_goodreads_url(self):
>> >              self.goodreads_url_edit.setText('')
>> >
>> >      def update_aliases(self):
>> > +        if 'goodreads.com' not in self.goodreads_url_edit.text():
>>
>> Ah, ok. It works if you have the full URL (ie. with https?):
>>
>> >>> urlparse.urlsplit("http://www.goodreads.com/foo")
>> SplitResult(scheme='http', netloc='www.goodreads.com', path='/foo', query='', fragment='')
>> >>> urlparse.urlsplit("http://goodreads.com/foo")
>> SplitResult(scheme='http', netloc='goodreads.com', path='/foo', query='', fragment='')
>> >>> urlparse.urlsplit("www.goodreads.com/foo")
>> SplitResult(scheme='', netloc='', path='www.goodreads.com/foo', query='', fragment='')
>> >>>
>>
>> —
>> You are receiving this because you modified the open/close state.
>>
>>
>> Reply to this email directly, view it on GitHub
>> <https://github.com/szarroug3/X-Ray_Calibre_Plugin/pull/55/files/91e68f6bfca6ed0c90477eb3f30704dc1b39fc70#r69025248>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AIEjUfzZm6gjNVmMkxoVr9273Zbkp3JUks5qQtrlgaJpZM4I-Mak>
>> .
>>
>